pkg/util/cluster/cluster.go

```diff
@@ -149,11 +149,15 @@ func (c *Cluster) Create(ctx context.Context, vnetResourceGroup, clusterName str
 	}
 
 	fpClientID := os.Getenv("AZURE_FP_CLIENT_ID")
-
-	fpSPID, err := c.getServicePrincipal(ctx, fpClientID)
-	if err != nil {
-		return err
+	var fpSPID string
+	fpSPID, ok := os.LookupEnv("AZURE_FP_SERVICE_PRINCIPAL_ID")
+	if !ok {
```

Comment:
This env var `AZURE_FP_SERVICE_PRINCIPAL_ID` is what we should be doing instead of having to check if it doesn't exist or not.  I think the idea of the fallback to the `getServicePrincipal` was to ensure some fail-safe way of providing the fpSPID if this isn't available (which shouldn't happen).  

As far as standardizing this approach across environments, I believe we should do that but the best method forward for that should be discussed further outside of this PR. @m1kola

Reply:
We could take out the lookup.  We always pass in this value, it should never be empty.